.guix/genenetwork3-package.scm

```diff
@@ -3,38 +3,51 @@
                 #:select (genenetwork3) #:prefix gn:)
   #:use-module ((gnu packages check) #:select (python-pylint))
   #:use-module ((gnu packages python-check) #:select (python-mypy))
+  #:use-module ((gnu packages linux) #:select (procps))
+  #:use-module ((gnu packages databases) #:select (virtuoso-ose))
   #:use-module (guix gexp)
   #:use-module (guix utils)
   #:use-module (guix git-download)
   #:use-module (guix packages))
 
 (define-public genenetwork3
   (package
-    (inherit gn:genenetwork3)
-    (version (string-append (package-version gn:genenetwork3) "-git"))
-    (source (local-file ".."
-                        "genenetwork3-checkout"
-                        #:recursive? #t
-                        #:select? (or (git-predicate (dirname (current-source-directory)))
-                                      (const #t))))))
+   (inherit gn:genenetwork3)
+   (version (string-append (package-version gn:genenetwork3) "-git"))
+   (source (local-file ".."
+                       "genenetwork3-checkout"
+                       #:recursive? #t
+                       #:select? (or (git-predicate (dirname (current-source-directory)))
+                                     (const #t))))))
 
 (define-public genenetwork3-all-tests
   (package
-    (inherit genenetwork3)
-    (arguments
-     (substitute-keyword-arguments (package-arguments genenetwork3)
-       ((#:phases phases #~%standard-phases)
-        #~(modify-phases #$phases
-            (add-before 'build 'pylint
-              (lambda _
-                (invoke "pylint" "main.py" "setup.py" "wsgi.py" "setup_commands"
-                        "tests" "gn3" "scripts" "sheepdog")))
-            (add-after 'pylint 'mypy
-              (lambda _
-                (invoke "mypy" ".")))))))
-    (native-inputs
-     (modify-inputs (package-native-inputs genenetwork3)
-       (prepend python-pylint)
-       (prepend python-mypy)))))
+   (inherit genenetwork3)
+   (arguments
+    (substitute-keyword-arguments (package-arguments genenetwork3)
+      ((#:phases phases #~%standard-phases)
+       #~(modify-phases #$phases
+	   (add-after
+	    'unpack 'patch-sources
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+	      (let ((virtuoso-ose (assoc-ref inputs "virtuoso-ose")))
+	        (substitute* "tests/fixtures/rdf.py"
+	         (("virtuoso-t")
+	          (string-append #$virtuoso-ose "/bin/virtuoso-t"))))))
+      	   (add-after 'build 'rdf-tests
+	   	   (lambda _
+	   	     (invoke "pytest" "-k" "rdf")))
+     	   (add-before 'build 'pylint
+      	     (lambda _
+      	       (invoke "pylint" "main.py" "setup.py" "wsgi.py" "setup_commands"
+      	     	      "tests" "gn3" "scripts" "sheepdog")))
+      	   (add-after 'pylint 'mypy
```

Comment:
@Alexanderlacuna Great question!  `add-before` and `add-after` essentially serve the same purpose, but the key difference is where in the build process you want to insert a new step.  In this case, we use `add-after` because we want to run `mypy` right `after pylint` but right before running the unit tests.  It’s a logical flow: we perform our static checks (like pylint and mypy) before running the unit tests to catch issues earlier.  As for the RDF tests, I’ll adjust them to run after the unit tests to maintain that order.

Reply:
I've addressed above [here](https://github.com/genenetwork/genenetwork3/pull/199)